crates/zkforge/bin/cmd/script/build.rs

```diff
@@ -1,33 +1,35 @@
 use super::*;
 use alloy_primitives::{Address, Bytes};
 use eyre::{Context, ContextCompat, Result};
-use foundry_cli::utils::get_cached_entry_by_name;
+
 use foundry_common::{
     compact_to_contract,
-    compile::{self, ContractSources},
+    compile::ContractSources,
     fs,
+    zk_compile::ZkSolc,
+    zksolc_manager::{setup_zksolc_manager, DEFAULT_ZKSOLC_VERSION},
 };
 use foundry_compilers::{
     artifacts::{CompactContractBytecode, ContractBytecode, ContractBytecodeSome, Libraries},
-    cache::SolFilesCache,
     contracts::ArtifactContracts,
     info::ContractInfo,
     ArtifactId, Project, ProjectCompileOutput,
 };
-use std::{collections::BTreeMap, str::FromStr};
+
+use std::{collections::BTreeMap, path::PathBuf, str::FromStr};
 use zkforge::link::{link_with_nonce_or_address, PostLinkInput, ResolvedDependency};
 
 impl ScriptArgs {
     /// Compiles the file or project and the verify metadata.
-    pub fn compile(&mut self, script_config: &mut ScriptConfig) -> Result<BuildOutput> {
+    pub async fn compile(&mut self, script_config: &mut ScriptConfig) -> Result<BuildOutput> {
```

Comment:
Changes in this file can be restored as we shouldnt be interacting with build.rs but rather zkbuild?